gallery_dl/extractor/photovogue.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.vogue.it/en/photovogue/"""
+
+from .common import Extractor, Message
+from datetime import datetime
+
+BASE_PATTERN = r"(?:https?://)?(?:www.vogue.it(?:/en)?/photovogue)"
+
+
+class PhotovogueUserExtractor(Extractor):
+    category = "photovogue"
+    subcategory = "user"
+    directory_fmt = ("{category}", "{photographer[id]}_{photographer[name]}")
+    filename_fmt = "{id}_{title}.{extension}"
+    archive_fmt = "{id}"
+    root = "https://www.vogue.it/en/photovogue/"
+    pattern = BASE_PATTERN + r"/portfolio/\?id=(\d+)"
+    test = (
+        ("https://www.vogue.it/en/photovogue/portfolio/?id=221252",),
+        ("https://www.vogue.it/photovogue/portfolio/?id=221252",),
+    )
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+        self.user_id = match.group(1)
+
+    def _photos(self):
+        page = 0
+
+        while True:
+            res = self.request(
+                "https://api.vogue.it/production/photos",
+                params={
+                    "count": 50,
+                    "order_by": "DESC",
+                    "page": page,
+                    "photographer_id": self.user_id,
+                },
+            ).json()
```

Comment:
This is just personal preference, but I'd prepare url and params before the loop, and then just call
`res = self.request(url, params=params).json()` inside the loop.

This would also get rid of the `page` variable, since you could just do `params["page"] += 1` instead.